langmeta-sqlite/src/main/resources/semanticdb.ddl

```diff
@@ -0,0 +1,49 @@
+create table document(
+  id int,
+  filename text,
+  contents text,
+  language text,
+  primary key (id)
+);
+
+create table name(
+  id int,
+  document int,
+  start int,
+  end int,
+  symbol int,
```

Comment:
I am concerned about making this a foreign key on symbol since it complicates insertions. For references, we typically don't have the denotation.

Reply:
How does it complicate insertions exactly?  FKs can be nullable so I am not understanding how it would matter whether we have a denotation or not.

Reply:
I'd like to avoid updates altogether, so if you have a reference to a symbol but no flag/signature of that symbol then you're out of luck. See other comment proposing to split "symbol" table into two tables

Reply:
This saves a lot of space, so I suggest we try to keep it as a FK. If we split symbol and denotation, this matter becomes moot anyway, right?

Reply:
It can be a FK as long as we split symbol/denotation :)

Reply:
👍